### DISCUSSION TITLE
How to render diferent pages with dynamic routes?

### DISCUSSION THREAD

**Discussion by aralroca** [vercel/next.js#25294] [Help]
How can I give a slug to render one page or another?

Example for `/content/[slug]`:

`/content/example1` - example page.
`/content/example2` - example page
`/content/another1` - another page

There is no defined pattern, but I have to ask an external API sending the `slug` as parameter to know what kind of landing it is and I have to render.

Thanks!

**Comment by PsyGik** [comment]
So there is one pattern which I used to solve this particular problem. The only caveat is that you must have some kind of a `template identifier`, i.e mapping b/w a slug and a component. 

Example of a mapping:

```jsx

/**
 * Import the templates dynamically so that Next.js creates bundles
 * for each template. We do not to load all the templates together
 */
const templates = {
  Example1: dynamic(() =>
    import(`@/templates/Example1`)
  ),
  Example2: dynamic(() =>
    import(`@/templates/Example2`)
  ),
  Another1: dynamic(() =>
    import(`@/templates/Another1`)
  ),
}

/**
 * Mapping between template and templateIdentifier.
 * value MUST match the `template` defined above
 */
const componentTemplateIdentifierMap = {
  example1: 'Example1',
  example2: 'Example2',
  another1: 'Another1'
}

```

Then in your `content/[slug]/index.js`, using your API get the value for `templateIdentifier`. Once you have that, based on the above mapping you can render either component. 

```jsx

 const getTemplateForIdentifier = (templateIdentifier) =>
    templates[componentTemplateIdentifierMap[templateIdentifier]]

  if (router.isFallback) {
    // First time on the route, data is being prepared
    return <div>loading...</div>
  } else {
    const Component = getTemplateForIdentifier(templateIdentifier)
    return (templateIdentifier && <Component />) || null
  }
```

You can also make use of `getStaticPaths` and `getStaticProps` to do [data fetching and render pages during build](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages)

**Reply by aralroca** [reply]
thanks, I will investigate on your advice. Thank you very much.
